h2o-docs/src/product/data-science/xgboost.rst

```diff
@@ -466,6 +470,15 @@ FAQs
 
     hadoop jar h2odriver.jar -nodes 1 -mapperXmx 20g -extramempercent 120
 
+- **When should I define the evalutation metric instead of letting H2O choose which metrics to calculate?**
+
+  While you don't always need to specify a custom ``eval_metric``, it is beneficial in two specific cases:
+
+    1. When H2O does not provide a suitable built-in metric (e.g. if you want to calculate classification error for a different threshold than the one automatically determined by H2O, you can do so by specifying ``eval_metric="error@<your threshold>"``);
+    2. When you have frequent scoring.
```

Comment:
Would be possilble to add an example here and illustrate  what is frequent scoring and what not. ([score_each_iteration](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/score_each_iteration.html), [score_tree_interval](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/score_tree_interval.html)  < THRESHOLD)

@valenad1 Please can you help to specify a THRESHOLD where `score_eval_metric_only` pays off? Are there any other parameters affecting scoring frequency?

Reply:
We can rewrite it to:
When you have frequent scoring. (e.g score_each_iteration=True, score_tree_interval < 10)

but its difficult to tell what it is... in the demo below it is specified as follow:

Case 2: Frequent scoring. By default H2O uses timing-triggered scoring, it is trying to make sure that majority of the time is used on model training as opposed to just model scoring. You can override this behavior and specify manually at what iterations you want model to be scored (eg. each iteration, every 5th iteration, ...). Because H2O calculates all possible metrics as opposed just few in native XGBoost, and needs to extract XGBoost model from the native memory, it can have a significant overhead when user desires to score very frequently. This can slow down the model building.

The upper bound in score_each_iteration=True and lower bound is score only the final model. Whatever is between depend on the time user have, number of trees, size of the tree.. 

IMHO typical usecase would be: 
Do you have score_each_iteration=True and do you want to keep it but only reduce training time? Try eval_metric..

Reply:
Thanks @valenad1, I think this is useful. A user will have at least a notion where a potential performance bottleneck might exist.

Reply:
I am good with that :)